stl/inc/algorithm

```diff
@@ -7405,6 +7609,339 @@ void inplace_merge(_ExPo&&, _BidIt _First, _BidIt _Mid, _BidIt _Last) noexcept /
+        const iter_difference_t<_It> _Count1, const iter_difference_t<_It> _Count2, iter_value_t<_It>* const _Temp_ptr,
+        const ptrdiff_t _Capacity, _Pr _Pred, _Pj _Proj, _It _Firstn, _It _Lastn, const iter_difference_t<_It> _Count1n,
+        const iter_difference_t<_It> _Count2n) {
+        // clang-format on
+        // common block of _Buffered_inplace_merge_divide_and_conquer, below
+        _It _Midn = _RANGES _Buffered_rotate_common(_Firstn, _Mid, _Lastn,
+            static_cast<iter_difference_t<_It>>(_Count1 - _Count1n), _Count2n, _Temp_ptr,
+            _Capacity); // rearrange middle
+        _RANGES _Buffered_inplace_merge_common(
+            _First, _Firstn, _Midn, _Count1n, _Count2n, _Temp_ptr, _Capacity, _Pred, _Proj); // merge each new part
+        _RANGES _Buffered_inplace_merge_common(_Midn, _Lastn, _Last,
+            static_cast<iter_difference_t<_It>>(_Count1 - _Count1n),
+            static_cast<iter_difference_t<_It>>(_Count2 - _Count2n), _Temp_ptr, _Capacity, _Pred, _Proj);
+    }
+
+    // clang-format off
+    template <bidirectional_iterator _It, class _Pr, class _Pj>
+        requires sortable<_It, _Pr, _Pj>
+    void _Buffered_inplace_merge_divide_and_conquer(_It _First, _It _Mid, _It _Last,
+        const iter_difference_t<_It> _Count1, const iter_difference_t<_It> _Count2, iter_value_t<_It>* const _Temp_ptr,
+        const ptrdiff_t _Capacity, _Pr _Pred, _Pj _Proj) {
+        // clang-format on
+        // merge sorted [_First, _Mid) with sorted [_Mid, _Last)
+        // usual invariants apply
+        if (_Count1 <= _Count2) {
+            const iter_difference_t<_It> _Count1n = _Count1 >> 1; // shift for codegen
+            const _It _Firstn                     = _RANGES next(_First, _Count1n);
+            const _It _Lastn =
+                _RANGES _Lower_bound_unchecked(_Mid, _Count1, _STD invoke(_Proj, *_Firstn), _Pred, _Proj);
+            const auto _Count2n = _RANGES distance(_Mid, _Lastn);
+            _RANGES _Buffered_inplace_merge_divide_and_conquer2(_STD move(_First), _STD move(_Mid), _STD move(_Last),
+                _Count1, _Count2, _Temp_ptr, _Capacity, _STD move(_Pred), _STD move(_Proj), _STD move(_Firstn),
+                _STD move(_Lastn), _Count1n, _Count2n);
+        } else {
+            const iter_difference_t<_It> _Count2n = _Count2 >> 1; // shift for codegen
+            const _It _Lastn                      = _RANGES next(_Mid, _Count2n);
+            const _It _Firstn =
+                _RANGES _Upper_bound_unchecked(_First, _Count2, _STD invoke(_Proj, *_Lastn), _Pred, _Proj);
+            const auto _Count1n = _RANGES distance(_First, _Firstn);
+            _RANGES _Buffered_inplace_merge_divide_and_conquer2(_STD move(_First), _STD move(_Mid), _STD move(_Last),
+                _Count1, _Count2, _Temp_ptr, _Capacity, _STD move(_Pred), _STD move(_Proj), _STD move(_Firstn),
+                _STD move(_Lastn), _Count1n, _Count2n);

```

Comment:
Notably these lines are identical to 7762-7764 _and_ the only callsites for `_Buffered_inplace_merge_divide_and_conquer2`. If you default-initialize the variables before and then assign them in the `if`/`else`, you could inline `_Buffered_inplace_merge_divide_and_conquer2` after.